consensus/spos/bn/bnSubroundsFactory.go

```diff
@@ -69,7 +69,9 @@ func checkNewFactoryParams(
 
 // SetAppStatusHandler method set appStatusHandler
 func (fct *factory) SetAppStatusHandler(handler core.AppStatusHandler) {
-	fct.appStatusHandler = handler
+	if handler != nil {
```

Comment:
The pattern already used in app status handlers impl is that this func should return error if the provided handler is nil. Let's keep this pattern.

Reply:
SetAppStatusHandler return error now